core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -592,6 +593,11 @@ private Builder(TableIdentifier ident, Schema schema, SessionContext context) {
       this.ident = ident;
       this.schema = schema;
       this.context = context;
+
+      // explicitly set defaults that apply only to new tables
+      propertiesBuilder.put(
+          TableProperties.PARQUET_COMPRESSION,
+          TableProperties.PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0);
```

Comment:
I don't think that this should be done for the REST catalog. The REST catalog implementation should use `TableMetadata.newTableMetadata` and that would apply the default, rather than setting it on the client and in the service.

I also think it's better if we avoid setting this in general (though I agree with the decision to set it by default) so it is better if we delegate to the service to apply it.

Reply:
Agreed, I'll update.

Reply:
Updated.